File: frontend/src/shared/contracts.ts
export type ErrorInfo = {
  error_code: string;
  message: string;
  summary: string;
  retryable: boolean;
  source: string;
  stage: string;
  metadata?: Record<string, string>;
};

export type ModelRef = {
  provider_id: string;
  model_id: string;
  display_name: string;
  enabled?: boolean;
  available?: boolean;
};

export type ProviderRef = {
  provider_id: string;
  display_name: string;
  base_url: string;
  chat_completions_path: string;
  models_path: string;
};

export type ProviderApplyStatus = "not_configured" | "saved_not_applied" | "applied";

export type ProviderTestStatus = "idle" | "success" | "error";

export type ProviderCredentialStatus = {
  has_credential: boolean;
  api_key_masked?: string;
  updated_at?: string;
  last_test_status?: ProviderTestStatus;
  last_test_message?: string;
  last_test_at?: string;
  apply_status: ProviderApplyStatus;
  applied_at?: string;
  pending_reload: boolean;
};

export type ProviderSettingsItem = ProviderRef & {
  credential_kind: string;
  supports_test: boolean;
  editable: boolean;
  credential_status: ProviderCredentialStatus;
};

export type ProviderSettingsResponse = {
  active_provider_id?: string;
  providers: ProviderSettingsItem[];
};

export type ProviderTestRequest = {
  provider_id: string;
  base_url?: string;
  api_key: string;
};

export type ProviderTestResponse = {
  ok: boolean;
  provider_id: string;
  message: string;
  checked_at?: string;
  error_code?: string;
};

export type ProviderSaveRequest = {
  provider_id: string;
  api_key: string;
};

export type ProviderSaveResponse = {
  ok: boolean;
  provider_id: string;
  message: string;
  credential_status: ProviderCredentialStatus;
};

export type ProviderApplyRequest = {
  provider_id: string;
};

export type ProviderApplyResponse = {
  ok: boolean;
  provider_id: string;
  message: string;
  apply_mode: string;
  applied_at?: string;
  restart_required: boolean;
};

export type ProviderRemoveRequest = {
  provider_id: string;
};

export type ProviderRemoveResponse = {
  ok: boolean;
  provider_id: string;
  message: string;
  state_code?: string;
};

export type WorkspaceRef = {
  workspace_id: string;
  name: string;
  root_path: string;
  is_active: boolean;
};

export type GitCommitSummary = {
  commit_id: string;
  short_message: string;
  author?: string;
  timestamp?: string;
};

export type GitSnapshot = {
  current_branch?: string;
  default_branch?: string;
  is_dirty: boolean;
  recent_commits?: GitCommitSummary[];
};

export type WorkspaceDocSummary = {
  path: string;
  kind: string;
  exists: boolean;
  summary: string;
  truncated: boolean;
};

export type CapabilitySpec = {
  capability_id: string;
  display_name: string;
  domain: string;
  risk_level: string;
  input_schema: string;
  output_kind: string;
  side_effect_level: string;
  supports_modes?: string[];
  verification_policy: string;
  connector_slot?: string;
  source_kind: string;
  requires_confirmation?: boolean;
};

export type CapabilityListResponse = {
  items: CapabilitySpec[];
};

export type ConnectorSlotSpec = {
  slot_id: string;
  display_name: string;
  priority: number;
  status: string;
  scope: string;
  current_capabilities?: string[];
  supported_actions?: string[];
  boundary: string;
  next_step: string;
};

export type ConnectorListResponse = {
  items: ConnectorSlotSpec[];
};

export type RepoContextSnapshot = {
  workspace_root: string;
  repo_root?: string;
  git_available: boolean;
  git_snapshot?: GitSnapshot;
  doc_summaries?: WorkspaceDocSummary[];
  warnings?: string[];
  collected_at: string;
};

export type SettingsResponse = {
  app_name: string;
  mode: string;
  model: ModelRef;
  available_models: ModelRef[];
  providers: ProviderRef[];
  workspace: WorkspaceRef;
  available_workspaces: WorkspaceRef[];
  approved_directories: DirectoryApproval[];
  directory_prompt_enabled: boolean;
  show_risk_level: boolean;
  ports: {
    gateway: number;
    runtime: number;
  };
  runtime_status: {
    ok: boolean;
    name: string;
    version: string;
  };
  memory_policy: MemoryPolicyStatus;
  diagnostics: DiagnosticsStatus;
  external_connections: ExternalConnectionSlot[];
};

export type MemoryPolicyStatus = {
  enabled: boolean;
  recall_strategy: string;
  write_strategy: string;
  cleanup_strategy: string;
  storage_root: string;
  sqlite_path: string;
  working_memory_dir: string;
  long_term_memory_path: string;
  knowledge_base_path: string;
  long_term_memory_count: number;
  knowledge_count: number;
  working_memory_files: number;
};

export type DiagnosticsStatus = {
  checked_at?: string;
  repo_root: string;
  storage_root: string;
  settings_path: string;
  run_log_path: string;
  event_log_path: string;
  runtime_reachable: boolean;
  runtime_version: string;
  provider_count: number;
  model_count: number;
  workspace_count: number;
  approved_directory_count: number;
  siyuan_root: string;
  siyuan_export_dir: string;
  siyuan_auto_write_enabled: boolean;
  siyuan_sync_enabled: boolean;
  warnings?: string[];
  errors?: string[];
};

export type ExternalConnectionSlot = {
  slot_id: string;
  display_name: string;
  priority: number;
  status: string;
  scope: string;
  current_tools: string[];
  supported_actions?: Array<"validate" | "recheck">;
  boundary: string;
  next_step: string;
};

export type ExternalConnectionActionRequest = {
  slot_id: string;
  action: "validate" | "recheck";
};

export type ExternalConnectionActionResponse = {
  slot_id: string;
  action: "validate" | "recheck";
  ok: boolean;
  message: string;
  updated_slot?: ExternalConnectionSlot;
  external_connections?: ExternalConnectionSlot[];
};

export type DiagnosticsCheckResponse = {
  checked_at: string;
  overall_ok: boolean;
  diagnostics: DiagnosticsStatus;
  warnings: string[];
  errors: string[];
};

export type DirectoryApproval = {
  approval_id: string;
  workspace_id: string;
  name: string;
  root_path: string;
  created_at?: string;
};

export type LogsResponse = {
  items: LogEntry[];
};

export type MemoryEntry = {
  id: string;
  kind: string;
  memory_kind?: string;
  title: string;
  summary: string;
  content: string;
  reason: string;
  scope: string;
  workspace_id: string;
  session_id: string;
  source_run_id: string;
  source: string;
  source_type: string;
  source_title?: string;
  source_event_type?: string;
  source_artifact_path?: string;
  governance_version?: string;
  governance_reason?: string;
  governance_source?: string;
  governance_at?: string;
  governance_status?: string;
  memory_action?: string;
  archive_reason?: string;
  verified: boolean;
  priority: number;
  archived: boolean;
  archived_at?: string;
  created_at: string;
  updated_at: string;
  timestamp: string;
};

export type MemoryListResponse = {
  items: MemoryEntry[];
};

export type ChatRunAccepted = {
  accepted: boolean;
  session_id: string;
  run_id: string;
  initial_status: string;
};

export type RuntimeRunRequest = {
  request_id: string;
  run_id: string;
  session_id: string;
  trace_id: string;
  user_input: string;
  mode: string;
  model_ref: ModelRef;
  provider_ref: ProviderRef;
  workspace_ref: WorkspaceRef;
  context_hints?: Record<string, string>;
  resume_from_checkpoint_id?: string;
  resume_strategy?: string;
  confirmation_decision?: ConfirmationDecision;
};

export type RunResult = {
  request_id?: string;
  run_id: string;
  session_id?: string;
  trace_id?: string;
  kind?: string;
  source?: string;
  status: string;
  final_answer: string;
  summary: string;
  error?: ErrorInfo;
  memory_write_summary?: string;
  final_stage: string;
  checkpoint_id?: string;
  resumable?: boolean;
};

export type RuntimeRunResponse = {
  events: RunEvent[];
  result: RunResult;
  confirmation_request?: ConfirmationRequest;
};

export type RunEvent = {
  event_id: string;
  kind?: string;
  source?: string;
  record_type?: string;
  source_type?: string;
  agent_id?: string;
  agent_label?: string;
  event_type: string;
  trace_id?: string;
  session_id: string;
  run_id: string;
  sequence: number;
  timestamp: string;
  stage: string;
  summary: string;
  detail?: string;
  tool_name?: string;
  tool_display_name?: string;
  tool_category?: string;
  output_kind?: string;
  result_summary?: string;
  artifact_path?: string;
  risk_level?: string;
  confirmation_id?: string;
  final_answer?: string;
  completion_status?: string;
  completion_reason?: string;
  verification_summary?: string;
  checkpoint_written?: boolean;
  context_snapshot?: RuntimeContextSnapshot;
  tool_call_snapshot?: ToolCallSnapshot;
  verification_snapshot?: VerificationSnapshot;
  metadata?: Record<string, string>;
};

export type RuntimeContextSnapshot = {
  workspace_root?: string;
  mode?: string;
  session_summary?: string;
  memory_digest?: string;
  knowledge_digest?: string;
  tool_preview?: string;
  reasoning_summary?: string;
  cache_status?: string;
  cache_reason?: string;
  assembly_profile?: string;
  includes_session?: boolean;
  includes_memory?: boolean;
  includes_knowledge?: boolean;
  includes_tool_preview?: boolean;
  prompt_static?: string;
  prompt_project?: string;
  prompt_dynamic?: string;
};

export type ToolCallSnapshot = {
  tool_name?: string;
  display_name?: string;
  category?: string;
  risk_level?: string;
  input_schema?: string;
  output_kind?: string;
  requires_confirmation?: boolean;
};

export type VerificationSnapshot = {
  code?: string;
  summary?: string;
  passed?: boolean;
  policy?: string;
  evidence?: string[];
};

export type LogEntry = {
  log_id: string;
  session_id: string;
  run_id: string;
  timestamp: string;
  level: string;
  category: string;
  source: string;
  record_type?: string;
  source_type?: string;
  agent_id?: string;
  agent_label?: string;
  event_type?: string;
  stage?: string;
  summary: string;
  detail?: string;
  tool_name?: string;
  tool_display_name?: string;
  tool_category?: string;
  output_kind?: string;
  result_summary?: string;
  artifact_path?: string;
  risk_level?: string;
  confirmation_id?: string;
  final_answer?: string;
  completion_status?: string;
  completion_reason?: string;
  verification_summary?: string;
  context_snapshot?: RuntimeContextSnapshot;
  tool_call_snapshot?: ToolCallSnapshot;
  verification_snapshot?: VerificationSnapshot;
  error?: ErrorInfo;
  metadata?: Record<string, string>;
};

export type ConfirmationRequest = {
  confirmation_id: string;
  run_id: string;
  risk_level: string;
  action_summary: string;
  reason: string;
  impact_scope: string;
  target_paths: string[];
  reversible: boolean;
  hazards: string[];
  alternatives: string[];
  kind: string;
};

export type ConfirmationDecision = {
  confirmation_id: string;
  run_id: string;
  decision: "approve" | "reject" | "cancel";
  note?: string;
  remember?: boolean;
};

export type ChatMessage = {
  id: string;
  role: "user" | "assistant";
  content: string;
  runId?: string;
};

File: gateway/internal/contracts/contracts.go
package contracts

import "local-agent/gateway/internal/config"

type ProviderRef struct {
	ProviderID          string `json:"provider_id"`
	DisplayName         string `json:"display_name"`
	BaseURL             string `json:"base_url"`
	ChatCompletionsPath string `json:"chat_completions_path"`
	ModelsPath          string `json:"models_path"`
	APIKey              string `json:"api_key"`
}

type ErrorInfo struct {
	ErrorCode string            `json:"error_code"`
	Message   string            `json:"message"`
	Summary   string            `json:"summary"`
	Retryable bool              `json:"retryable"`
	Source    string            `json:"source"`
	Stage     string            `json:"stage"`
	Metadata  map[string]string `json:"metadata,omitempty"`
}

type RunRequest struct {
	RequestID            string                `json:"request_id"`
	RunID                string                `json:"run_id"`
	SessionID            string                `json:"session_id"`
	TraceID              string                `json:"trace_id"`
	UserInput            string                `json:"user_input"`
	Mode                 string                `json:"mode"`
	ModelRef             config.ModelRef       `json:"model_ref"`
	ProviderRef          ProviderRef           `json:"provider_ref"`
	WorkspaceRef         config.WorkspaceRef   `json:"workspace_ref"`
	ContextHints         map[string]string     `json:"context_hints,omitempty"`
	ResumeFromCheckpointID string              `json:"resume_from_checkpoint_id,omitempty"`
	ResumeStrategy       string                `json:"resume_strategy,omitempty"`
	ConfirmationDecision *ConfirmationDecision `json:"confirmation_decision,omitempty"`
}

type RuntimeContextSnapshot struct {
	WorkspaceRoot    string `json:"workspace_root,omitempty"`
	Mode             string `json:"mode,omitempty"`
	SessionSummary   string `json:"session_summary,omitempty"`
	MemoryDigest     string `json:"memory_digest,omitempty"`
	KnowledgeDigest  string `json:"knowledge_digest,omitempty"`
	ToolPreview      string `json:"tool_preview,omitempty"`
	ReasoningSummary string `json:"reasoning_summary,omitempty"`
	CacheStatus      string `json:"cache_status,omitempty"`
	CacheReason      string `json:"cache_reason,omitempty"`
	AssemblyProfile  string `json:"assembly_profile,omitempty"`
	IncludesSession  bool   `json:"includes_session,omitempty"`
	IncludesMemory   bool   `json:"includes_memory,omitempty"`
	IncludesKnowledge bool  `json:"includes_knowledge,omitempty"`
	IncludesToolPreview bool `json:"includes_tool_preview,omitempty"`
	PromptStatic     string `json:"prompt_static,omitempty"`
	PromptProject    string `json:"prompt_project,omitempty"`
	PromptDynamic    string `json:"prompt_dynamic,omitempty"`
}

type ToolCallSnapshot struct {
	ToolName             string `json:"tool_name,omitempty"`
	DisplayName          string `json:"display_name,omitempty"`
	Category             string `json:"category,omitempty"`
	RiskLevel            string `json:"risk_level,omitempty"`
	InputSchema          string `json:"input_schema,omitempty"`
	OutputKind           string `json:"output_kind,omitempty"`
	RequiresConfirmation bool   `json:"requires_confirmation,omitempty"`
	ArgumentsJSON        string `json:"arguments_json,omitempty"`
}

type VerificationSnapshot struct {
	Code    string `json:"code,omitempty"`
	Summary string `json:"summary,omitempty"`
	Passed  bool   `json:"passed,omitempty"`
	Policy  string `json:"policy,omitempty"`
	Evidence []string `json:"evidence,omitempty"`
}

type RepoContextSnapshot struct {
	WorkspaceRoot string                `json:"workspace_root"`
	RepoRoot      *string               `json:"repo_root,omitempty"`
	GitAvailable  bool                  `json:"git_available"`
	GitSnapshot   *GitSnapshot          `json:"git_snapshot,omitempty"`
	DocSummaries  []WorkspaceDocSummary `json:"doc_summaries,omitempty"`
	Warnings      []string              `json:"warnings,omitempty"`
	CollectedAt   string                `json:"collected_at"`
}

type GitSnapshot struct {
	CurrentBranch *string            `json:"current_branch,omitempty"`
	DefaultBranch *string            `json:"default_branch,omitempty"`
	IsDirty       bool               `json:"is_dirty"`
	RecentCommits []GitCommitSummary `json:"recent_commits,omitempty"`
}

type GitCommitSummary struct {
	CommitID     string  `json:"commit_id"`
	ShortMessage string  `json:"short_message"`
	Author       *string `json:"author,omitempty"`
	Timestamp    *string `json:"timestamp,omitempty"`
}

type WorkspaceDocSummary struct {
	Path      string `json:"path"`
	Kind      string `json:"kind"`
	Exists    bool   `json:"exists"`
	Summary   string `json:"summary"`
	Truncated bool   `json:"truncated"`
}

type CapabilitySpec struct {
	CapabilityID         string   `json:"capability_id"`
	DisplayName          string   `json:"display_name"`
	Domain               string   `json:"domain"`
	RiskLevel            string   `json:"risk_level"`
	InputSchema          string   `json:"input_schema"`
	OutputKind           string   `json:"output_kind"`
	SideEffectLevel      string   `json:"side_effect_level"`
	SupportsModes        []string `json:"supports_modes,omitempty"`
	VerificationPolicy   string   `json:"verification_policy"`
	ConnectorSlot        string   `json:"connector_slot,omitempty"`
	SourceKind           string   `json:"source_kind"`
	RequiresConfirmation bool     `json:"requires_confirmation,omitempty"`
}

type CapabilityListResponse struct {
	Items []CapabilitySpec `json:"items"`
}

type ConnectorSlotSpec struct {
	SlotID              string   `json:"slot_id"`
	DisplayName         string   `json:"display_name"`
	Priority            int      `json:"priority"`
	Status              string   `json:"status"`
	Scope               string   `json:"scope"`
	CurrentCapabilities []string `json:"current_capabilities,omitempty"`
	SupportedActions    []string `json:"supported_actions,omitempty"`
	Boundary            string   `json:"boundary"`
	NextStep            string   `json:"next_step"`
}

type ConnectorListResponse struct {
	Items []ConnectorSlotSpec `json:"items"`
}

type RunResult struct {
	RequestID          string     `json:"request_id,omitempty"`
	RunID              string     `json:"run_id"`
	SessionID          string     `json:"session_id,omitempty"`
	TraceID            string     `json:"trace_id,omitempty"`
	Kind               string     `json:"kind,omitempty"`
	Source             string     `json:"source,omitempty"`
	Status             string     `json:"status"`
	FinalAnswer        string     `json:"final_answer"`
	Summary            string     `json:"summary"`
	Error              *ErrorInfo `json:"error,omitempty"`
	MemoryWriteSummary *string    `json:"memory_write_summary,omitempty"`
	FinalStage         string     `json:"final_stage"`
	CheckpointID       *string    `json:"checkpoint_id,omitempty"`
	Resumable          *bool      `json:"resumable,omitempty"`
}

type RunEvent struct {
	EventID              string                  `json:"event_id"`
	Kind                 string                  `json:"kind,omitempty"`
	Source               string                  `json:"source,omitempty"`
	RecordType           string                  `json:"record_type,omitempty"`
	SourceType           string                  `json:"source_type,omitempty"`
	AgentID              string                  `json:"agent_id,omitempty"`
	AgentLabel           string                  `json:"agent_label,omitempty"`
	EventType            string                  `json:"event_type"`
	TraceID              string                  `json:"trace_id,omitempty"`
	SessionID            string                  `json:"session_id"`
	RunID                string                  `json:"run_id"`
	Sequence             int                     `json:"sequence"`
	Timestamp            string                  `json:"timestamp"`
	Stage                string                  `json:"stage"`
	Summary              string                  `json:"summary"`
	Detail               string                  `json:"detail,omitempty"`
	ToolName             string                  `json:"tool_name,omitempty"`
	ToolDisplayName      string                  `json:"tool_display_name,omitempty"`
	ToolCategory         string                  `json:"tool_category,omitempty"`
	OutputKind           string                  `json:"output_kind,omitempty"`
	ResultSummary        string                  `json:"result_summary,omitempty"`
	ArtifactPath         string                  `json:"artifact_path,omitempty"`
	RiskLevel            string                  `json:"risk_level,omitempty"`
	ConfirmationID       string                  `json:"confirmation_id,omitempty"`
	FinalAnswer          string                  `json:"final_answer,omitempty"`
	CompletionStatus     string                  `json:"completion_status,omitempty"`
	CompletionReason     string                  `json:"completion_reason,omitempty"`
	VerificationSummary  string                  `json:"verification_summary,omitempty"`
	CheckpointWritten    bool                    `json:"checkpoint_written,omitempty"`
	ContextSnapshot      *RuntimeContextSnapshot `json:"context_snapshot,omitempty"`
	ToolCallSnapshot     *ToolCallSnapshot       `json:"tool_call_snapshot,omitempty"`
	VerificationSnapshot *VerificationSnapshot   `json:"verification_snapshot,omitempty"`
	Metadata             map[string]string       `json:"metadata,omitempty"`
}

type ConfirmationRequest struct {
	ConfirmationID string   `json:"confirmation_id"`
	RunID          string   `json:"run_id"`
	RiskLevel      string   `json:"risk_level"`
	ActionSummary  string   `json:"action_summary"`
	Reason         string   `json:"reason"`
	ImpactScope    string   `json:"impact_scope"`
	TargetPaths    []string `json:"target_paths"`
	Reversible     bool     `json:"reversible"`
	Hazards        []string `json:"hazards"`
	Alternatives   []string `json:"alternatives"`
	Kind           string   `json:"kind"`
}

type ConfirmationDecision struct {
	ConfirmationID string `json:"confirmation_id"`
	RunID          string `json:"run_id"`
	Decision       string `json:"decision"`
	Note           string `json:"note,omitempty"`
	Remember       bool   `json:"remember,omitempty"`
}

type LogEntry struct {
	LogID                string                  `json:"log_id"`
	SessionID            string                  `json:"session_id"`
	RunID                string                  `json:"run_id"`
	Timestamp            string                  `json:"timestamp"`
	Level                string                  `json:"level"`
	Category             string                  `json:"category"`
	Source               string                  `json:"source"`
	RecordType           string                  `json:"record_type,omitempty"`
	SourceType           string                  `json:"source_type,omitempty"`
	AgentID              string                  `json:"agent_id,omitempty"`
	AgentLabel           string                  `json:"agent_label,omitempty"`
	EventType            string                  `json:"event_type,omitempty"`
	Stage                string                  `json:"stage,omitempty"`
	Summary              string                  `json:"summary"`
	Detail               string                  `json:"detail,omitempty"`
	ToolName             string                  `json:"tool_name,omitempty"`
	ToolDisplayName      string                  `json:"tool_display_name,omitempty"`
	ToolCategory         string                  `json:"tool_category,omitempty"`
	OutputKind           string                  `json:"output_kind,omitempty"`
	ResultSummary        string                  `json:"result_summary,omitempty"`
	ArtifactPath         string                  `json:"artifact_path,omitempty"`
	RiskLevel            string                  `json:"risk_level,omitempty"`
	ConfirmationID       string                  `json:"confirmation_id,omitempty"`
	FinalAnswer          string                  `json:"final_answer,omitempty"`
	CompletionStatus     string                  `json:"completion_status,omitempty"`
	CompletionReason     string                  `json:"completion_reason,omitempty"`
	VerificationSummary  string                  `json:"verification_summary,omitempty"`
	ContextSnapshot      *RuntimeContextSnapshot `json:"context_snapshot,omitempty"`
	ToolCallSnapshot     *ToolCallSnapshot       `json:"tool_call_snapshot,omitempty"`
	VerificationSnapshot *VerificationSnapshot   `json:"verification_snapshot,omitempty"`
	Error                *ErrorInfo              `json:"error,omitempty"`
	Metadata             map[string]string       `json:"metadata,omitempty"`
}

type RuntimeRunResponse struct {
	Events              []RunEvent           `json:"events"`
	Result              RunResult            `json:"result"`
	ConfirmationRequest *ConfirmationRequest `json:"confirmation_request,omitempty"`
}

type MemoryEntry struct {
	ID                 string `json:"id"`
	Kind               string `json:"kind"`
	MemoryKind         string `json:"memory_kind,omitempty"`
	Title              string `json:"title"`
	Summary            string `json:"summary"`
	Content            string `json:"content"`
	Reason             string `json:"reason"`
	Scope              string `json:"scope"`
	WorkspaceID        string `json:"workspace_id"`
	SessionID          string `json:"session_id"`
	SourceRunID        string `json:"source_run_id"`
	Source             string `json:"source"`
	SourceType         string `json:"source_type"`
	SourceTitle        string `json:"source_title,omitempty"`
	SourceEventType    string `json:"source_event_type,omitempty"`
	SourceArtifactPath string `json:"source_artifact_path,omitempty"`
	GovernanceVersion  string `json:"governance_version,omitempty"`
	GovernanceReason   string `json:"governance_reason,omitempty"`
	GovernanceSource   string `json:"governance_source,omitempty"`
	GovernanceAt       string `json:"governance_at,omitempty"`
	GovernanceStatus   string `json:"governance_status,omitempty"`
	MemoryAction       string `json:"memory_action,omitempty"`
	ArchiveReason      string `json:"archive_reason,omitempty"`
	Verified           bool   `json:"verified"`
	Priority           int    `json:"priority"`
	Archived           bool   `json:"archived"`
	ArchivedAt         string `json:"archived_at,omitempty"`
	CreatedAt          string `json:"created_at"`
	UpdatedAt          string `json:"updated_at"`
	Timestamp          string `json:"timestamp"`
}

type MemoryListResponse struct {
	Items []MemoryEntry `json:"items"`
}
